src/components/ArticlePage/MarkdownHtml.js

```diff
@@ -9,35 +9,73 @@ import {
 } from '../../config';
 import { H4, H5, H6, P, Ul, Ol, Li, Hr, A, Strong, Em } from '../../components';
 
-require('prismjs/themes/prism-solarizedlight.css');
+require('prismjs/themes/prism-tomorrow.css');
 
 const HighlightStyle = createGlobalStyle`
+
+  // Decrease the distance between p and the code block
+  p + .gatsby-highlight {
+    margin-top: -${2 * baselineSmall}px;
+
+    @media (min-width: ${baselineBreakpoint}px) {
+      margin-top: -${2 * baselineLarge}px;
+    }
+  }
+
+  li p + .gatsby-highlight {
+
+    // Baseline offsets
+    margin-top: 0;
+    margin-bottom: 18px;
+
+    @media (min-width: ${baselineBreakpoint}px) {
+      // Baseline offset
+      top: -2px;
+
+      position: relative;
+      margin-bottom: 16px;
+    }
+  }
+
   .gatsby-highlight  {
+    margin-bottom: ${2 * baselineSmall}px;
+
+    @media (min-width: ${baselineBreakpoint}px) {
+      margin-bottom: ${2 * baselineLarge}px;
+    }
 
     // Remove theme default styles
     pre[class*="language-"] {
-      background-color: transparent;
       margin: ${baselineSmall}px 0;
-      padding: ${2 * baselineSmall}px 0;
+
+      // Baseline offset
+      padding: ${2 * baselineSmall - 1}px 16px ${2 * baselineSmall}px 16px;
```

Comment:
This does seem to fix code blocks with a single line, but block with multiple lines still break the baseline alignment on mobile:

<img width="446" alt="Screen Shot 2019-04-01 at 16 33 39" src="https://user-images.githubusercontent.com/53923/55331552-3148d980-549c-11e9-9953-e9e7c4f5123d.png">

Maybe play around with line-height?